src/ResourceManager/Sql/Commands.Sql/Failover Group/Cmdlet/NewAzureSqlFailoverGroup.cs

```diff
@@ -0,0 +1,162 @@
+    {
+        /// <summary>
+        /// Gets or sets the name of the Failover Group to create.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The name of the Azure SQL FailoverGroup to create.")]
+        [ValidateNotNullOrEmpty]
+        public string FailoverGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the partner resource group name for Azure SQL Failover Group
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The partner resource group name for Azure SQL Database Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PartnerResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the partner server name for Azure SQL Failover Group
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The partner server name for Azure SQL Database Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PartnerServerName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the failover policy without data loss for the Sql Azure Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The failover policy without data loss for the failover group.")]
+        [ValidateNotNullOrEmpty]
+        public FailoverPolicy FailoverPolicy { get; set; }
+
+        /// <summary>
+        /// Gets or sets the grace period with data loss for the Sql Azure Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The grace period for failover with data loss of the failover group.")]
+        [ValidateNotNullOrEmpty]
+        public int GracePeriodWithDataLossHour { get; set; }
+
+        /// <summary>
+        /// Gets or sets the failover policy for read only endpoint of theSql Azure Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The failover policy for read only endpoint of the failover group.")]
+        [ValidateNotNullOrEmpty]
+        public AllowReadOnlyFailoverToPrimary AllowReadOnlyFailoverToPrimary { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tag associated with the Azure Sql Failover Group
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The tag to associate with the Azure Sql Failover Group")]
+        [Alias("Tag")]
+        public Hashtable Tag { get; set; }
+
+        /// <summary>
+        /// Overriding to add warning message
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+        }
+
+        /// <summary>
+        /// Get the entities from the service
+        /// </summary>
+        /// <returns>The list of entities</returns>
+        protected override IEnumerable<AzureSqlFailoverGroupModel> GetEntity()
```

Comment:
@ziwa-msft what is the purpose of this method? It appears that it either returns null or throws